### ISSUE TITLE
Change detection with sorting doesn't fire when remote changes happen

### ISSUE THREAD

**Issue by mmwo** [derek-fong/angular2-meteor-boilerplate#4]
Thanks for you great example of angular2-meteor.

Anyway I found an issue with sorting that was present from the old angular2-meteor apps.
Step to reproduce:
1. Open two windows of the application  on the demo list view
2. From the 1st window edit some demo and enter cost that will cause changing order of the demos on the list.
3. The change is tracked on the 1st window but 2nd window complains that it tries to get _id of undefined.

**Edit:**
The code in the service works and partially solves my issue

```
  get allDemos$(): ObservableCursor<DemoItem[]> {
    this.subscribe('allDemos');
    return Demos.find({}, { sort: { cost: -1 } }).zone();
  }
```

with autorun wrapper in component 

```
this.autorun(()=>{
      this.demoItems$ = this.demoService.allDemos$;
    })
```

However have no idea what happens here. 
The order changes only in the window that action is made. At least the second window doesn't crash
Probably I just broke the breaking part, not fixed it... =D

**Comment by derek-fong** [comment]
@Jitra yes you are right I have tested and confirmed updating `DemoDetail` will not trigger `DemoList` to update sort. 

I can see people have reported similar issues but not sure if those issues are related to this - Got a deadline for another project atm but I will look into this ASAP.
